minique/work/worker.py

```diff
@@ -56,7 +59,11 @@ def tick(self) -> Optional[Job]:
         if job.status == JobStatus.CANCELLED:  # Simply skip running cancelled jobs
             return None
         runner = self.job_runner_class(worker=self, job=job)
-        runner.run()
+        heart = JobPacemakerThread(self.heart_rate, job)  # post a heartbeat once a minute
+        heart.start()
+        with closing(heart):
+            runner.run()
+        heart.join()
```

Comment:
Re the earlier comment about moving the thread implementation to tests, this would also be snipped out.

```suggestion
```